File: Two_Sum_II_Input_array_is_sorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int start = 0, end = numbers.length - 1;
        while (start < end) {
            int startVal = numbers[start];
            int endVal = numbers[end];
            int curVal = startVal + endVal;
            if (curVal == target) {
                return new int[] {start + 1, end + 1};
            }
            else if(curVal > target) {
                end--;
            } else {
                start++;
            }
        }
        return new int[]{};
    }
}

File: TwoSumII-InputArrayIsSorted.go
package main

import "fmt"

func twoSum(numbers []int, target int) []int {
	i := 0
	j := len(numbers) - 1

	for i < j {
		sum := numbers[i] + numbers[j]
		if sum == target {
			return []int{i + 1, j + 1}
		} else if sum > target {
			j--
		} else {
			i++
		}
	}
	return []int{}
}

func main() {
	fmt.Print(twoSum([]int{2, 7, 11, 15}, 9))
}
